src/controllers/wallet/p2sh/tx-proposal.controller.js

```diff
@@ -48,6 +50,10 @@ async function buildTxProposal(req, res) {
       changeAddress,
     });
     const txData = await sendTransaction.prepareTxData();
+
+    if (markAsUsed) {
+      await sendTransaction.updateOutputSelected(true);
+    }
```

Comment:
Does it need another PR from the lib? I might be wrong but seeing the flow from the lib, the `updateOutputSelected` checks `if (this.transaction === null) {` and, in this case, it should be `null`.

```
const sendTransaction = new SendTransaction({
      storage: req.wallet.storage,
      outputs,
      inputs,
      changeAddress,
    });
    const txData = await sendTransaction.prepareTxData();
```

You instantiate the object with `outputs` and `inputs` and the method `prepareTxData` doesn't set `this.transaction`